### ISSUE TITLE
Release 1.12.0

### ISSUE THREAD

**Issue by enricoros** [enricoros/big-AGI#363]
## Release checklist:

- [x] Create a new [Release Issue](https://github.com/enricoros/big-AGI/issues/new?assignees=enricoros&projects=enricoros/4&template=maintainers-release.md&title=Release+1.12.0)
  - [x] Replace 1.11.0 with the _former_ release, and _1.12.0_ with THIS
- [x] Update the [Roadmap](https://github.com/users/enricoros/projects/4/views/2) calling out shipped features
- [x] Create and update a [Milestone](https://github.com/enricoros/big-agi/milestones) for the release
  - [x] Assign this task
  - [x] Assign all the shipped roadmap Issues
  - [x] Assign the relevant [recently closed Isssues](https://github.com/enricoros/big-agi/issues?q=is%3Aclosed+sort%3Aupdated-desc)
- Code changes:
  - [x] Create a release branch 'release-x.y.z': `git checkout -b release-1.12.0`
  - [x] Create a temporary tag `git tag v1.12.0 && git push opensource --tags`
  - [x] Create a [New Draft GitHub Release](https://github.com/enricoros/big-agi/releases/new), and generate the automated changelog (for new contributors)
  - [x] Update the release version in package.json, and `npm i`
  - [x] Update in-app News [src/apps/news/news.data.tsx](/src/apps/news/news.data.tsx)
  - [x] Update the in-app News version number
  - [x] Update the README.md with the new release
  - [x] Copy the highlights to the [docs/changelog.md](/docs/changelog.md)
- Release:
  - [x] merge onto main `git checkout main && git merge --no-ff release-1.12.0`
  - [x] re-tag `git tag -f v1.12.0 && git push opensource --tags -f'
  - [x] verify deployment on Vercel
  - [x] verify container on GitHub Packages
  - [x] update the GitHub release
  - [x] push as stable `git push opensource main:main-stable`
- Announce:
  - [x] Discord announcement
  - [x] Twitter announcement

### Links

- Milestone: https://github.com/enricoros/big-AGI/milestone/12
- GitHub release: https://github.com/enricoros/big-AGI/releases/tag/v1.12.0
- Former release task: #335

## Artifacts Generation

```markdown
You help me generate the following collateral for the new release of my opensource application
called big-AGI. The new release is 1.12.0.
To familiarize yourself with the application, the following are the Website and the GitHub README.md.
```

- paste the URL: https://big-agi.com
- drag & drop: [README.md](https://raw.githubusercontent.com/enricoros/big-AGI/main/README.md)

```markdown
I am announcing a new version, 1.12.0.
For reference, the following was the collateral for 1.11.0 (Discord announcement,
GitHub Release, in-app-news file news.data.tsx, changelog.md).
```

- paste the former: `discord announcement`,
- `GitHub release`,
- `news.data.tsx`,
- `changelog.md`

```markdown
The following are the new developments for 1.12.0:
```

- paste the link to the milestone (closed) and each individual issue (content will be downloaded)
- paste the git changelog  `git log --pretty=format:"%h %an %B" v1.11.0..v1.12.0 | clip`

### news.data.tsx

```markdown
I need the following from you:

1. a table summarizing all the new features in 1.12.0 (description, significance, usefulness, do not link the commit, but have the issue number), which will be used for the artifacts later
2. after the table score each feature from a user impact and magnitude point of view
3. Improve the table, in decreasing order of importance for features, fixing any detail that's missing, in particular check if there are commits of significance from a user or developer point of view, which are not contained in the table
4. I want you then to update the news.data.tsx for the new release
```

### Readme (and Changelog)

```markdown
I need you to update the README.md and the with the new release.
Attaching the in-app news, with my language for you to improve on, but keep the tone.
```

### GitHub release

```markdown
Please create the 1.12.0 Release Notes for GitHub.
Use a truthful and honest tone, understanding that people's time and attention span is short.
Today is 2024-1-1.
```

Now paste-attachment the former release notes (or 1.5.0 which was accurate and great), including the new contributors and
some stats (# of commits, etc.), and roll it for the new release.

### Discord announcement

```markdown
Can you generate my 1.12.0 big-AGI discord announcement from the GitHub Release announcement?
Please keep the formatting and stye of the discord announcement for 1.11.0, but with the new messaging above.
```

**Comment by enricoros** [comment]
More misc info:
 - Call App
 - Mobile UI
 - updated tiktoken
 - large speed improvements
 - updated ollama models
 - 150 commits, 7,000+ lines changed

**Comment by enricoros** [comment]
Screenshots
---

https://github.com/enricoros/big-AGI/assets/32999/95ceb03c-945d-4fdd-9a9f-3317beb54f3f

**Comment by enricoros** [comment]
news.data.tsx
---
```
  {// 🆕
    versionCode: '1.12.0',
    versionName: 'AGI Hotline',
    versionMoji: '✨🗣️',
    versionDate: new Date('2024-01-26T12:30:00Z'),
    items: [
      { text: <><B issue={354}>Voice Call Personas</B>: save time, recap conversations</>, issue: 354 },
      { text: <>Updated <B issue={364}>OpenAI Models</B> to the 0125 release</>, issue: 364 },
      { text: <>Chats: Auto-<B issue={222}>Rename</B> and <B issue={360}>assign folders</B></>, issue: 222 },
      { text: <><B issue={356}>Link Sharing</B> makeover and control</>, issue: 356 },
      { text: <><B issue={358}>Accessibility</B> for screen readers</>, issue: 358 },
      { text: <>Export chats to <B>Markdown</B></>, issue: 337 },
      { text: <>Paste <B>tables from Excel</B></>, issue: 286 },
      { text: <>Large optimizations</> },
      { text: <>Ollama updates</>, issue: 309 },
      { text: <>Over <B>150 commits</B> and <B>7,000+ lines changed</B> for development enhancements</>, dev: true },
    ],
  },
```

**Comment by enricoros** [comment]
Readme.md
---
```
## What's New in 1.12.0 · Jan 26, 2024 · AGI Hotline

https://github.com/enricoros/big-AGI/assets/32999/95ceb03c-945d-4fdd-9a9f-3317beb54f3f

- **Voice Calls**: real-time voice call your personas out of the blue or in relation to a chat [#354](https://github.com/enricoros/big-AGI/issues/354)
- Support **OpenAI 0125** Models. [#364](https://github.com/enricoros/big-AGI/issues/364)
- Rename or Auto-Rename chats.  [#222](https://github.com/enricoros/big-AGI/issues/222), [#360](https://github.com/enricoros/big-AGI/issues/360)
- More control over **Link Sharing** [#356](https://github.com/enricoros/big-AGI/issues/356)
- **Accessibility** to screen readers [#358](https://github.com/enricoros/big-AGI/issues/358)
- Export chats to Markdown [#337](https://github.com/enricoros/big-AGI/issues/337)
- Paste tables from Excel [#286](https://github.com/enricoros/big-AGI/issues/286)
- Ollama model updates and context window detection fixes [#309](https://github.com/enricoros/big-AGI/issues/309)

````

**Comment by enricoros** [comment]
GitHub Announcement
---
# 🚀 big-AGI 1.12.0 - AGI Hotline

Introducing big-AGI 1.12.0, a significant step in refining AGI interaction and management. Who can it be now? Your phone is ringing - it's your AGI. Voice Call, with new AI models, a snappier and juicier UI, and more. Here's the essence of what's new:
- **Voice Calls**: real-time voice call your personas out of the blue or in relation to a chat #354 
- **Updated OpenAI 0125** Models. #364 
- Rename and **Auto-Rename** chats.  #222 
- **Link Sharing Revamped**, with more user control #356 
- **Accessibility** to screen readers #358
- Export chats to Markdown #337 
- Paste tables from Excel #286 
- Ollama model updates and context window detection fixes #309 

For Developers:
- Over 150 commits and 7,000+ lines changed, emphasizing performance optimizations and under-the-hood improvements for a smoother experience.
- Dive into the details: [v1.11.0...v1.12.0](https://github.com/enricoros/big-AGI/compare/v1.11.0...v1.12.0)

### Upgrade Now
Upgrade to big-AGI 1.12.0 today (code, docker, or the hosted site).

Join us on [Discord](https://discord.gg/MkH4qj2Jp9) and explore our [public roadmap](https://github.com/users/enricoros/projects/4/views/2). Together, we're shaping its future! 🚀

**Comment by enricoros** [comment]
Discord Annoucement
---
@everyone ## :sparkles: **big-AGI 1.12.0 - AGI Hotline (Bling)** Released  

Massive drop :partyblob: . Update to 1.12.0 and get closer to your AGI personas than ever before:

- **Voice Calls**: Dial-up your AGI personas for real-time calls, out of the blue or over a chat
- **Updated OpenAI Models**: just announced by OpenAI the 0125 models
- Chats renaming, **auto-renaming**, exporting and folder selections
- Link Sharing makeover: share with more control
- Accessibility for screen readers - fully reforged the structure
- Paste from Excel, and export to Markdown, 
- Ollama models and context updates

*For Developers*: This release is larger than usual, with over 150 commits and 7,000+ lines changed, in features, React optimizations, and fixes across the board :capybarathink:

*Quick Links*: 
- Update now at https://get.big-agi.com or grab the latest Docker image https://github.com/enricoros/big-AGI/pkgs/container/big-agi
- Dive into what's next on our roadmap: https://github.com/users/enricoros/projects/4/views/2

:rocket: **So much more coming, and lots of exciting innovation - ping the devs, join us, so much to do**! :partyblob:
